src/mame/drivers/ymdx100.cpp

```diff
@@ -92,20 +283,258 @@ void yamaha_dx100_state::mem_map(address_map &map)
 }
 
 static INPUT_PORTS_START(dx100)
+	PORT_START("P2")
+	// TODO should 0x02, 0x04, 0x10, and 0x80 be listed here? they should be handled by the other interconnections in this file
+	// TODO if so, verify the active states of the MIDI ports
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_OUTPUT )   PORT_NAME("LED") PORT_WRITE_LINE_MEMBER(yamaha_dx100_state, led_w)
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_UNUSED )  // tied to ground
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_CUSTOM )  // 500khz clock
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_CUSTOM )  PORT_CUSTOM_MEMBER(yamaha_dx100_state, midi_in_r)
+	PORT_BIT( 0x10, IP_ACTIVE_HIGH, IPT_OUTPUT )  // MIDI out
+	PORT_CONFNAME( 0x20, 0x00, "Foot Switch Connected?" )
+	PORT_CONFSETTING( 0x00, "Connected" )
+	PORT_CONFSETTING( 0x20, "Disconnected" )
+	PORT_BIT( 0x40, IP_ACTIVE_HIGH, IPT_OTHER )   PORT_NAME("Foot Switch")
+	PORT_BIT( 0x80, IP_ACTIVE_HIGH, IPT_UNUSED )  // tied to ground
+
+	PORT_START("P6_0")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_OTHER )  PORT_NAME("Data Entry +1/Yes/On")
```

Comment:
Giving `IPT_OTHER` has the side effect that no inputs are mapped, needing a manual user intervention remapping on startup.
Consider giving a `PORT_CODE(KEYCODE_*)` host keyboard default for at least the most commonly used inputs, more or less based off the original synth input layout.

Reply:
Yeah, fully mapping these to a PC keyboard is non-trivial and likely need user feedback afterwards. Regardless it is a nice to have for any quick testing scenario.

By quick checking some synths uses `IPT_KEYBOARD` in place of `IPT_OTHER` for convenience in having the UI scroll lock mechanism

Reply:
Can the IPT_KEYBOARD keys have their own set of keycodes for the UI scroll lock mechanism, or do they just share the same mechanisms? I mapped all the non-keyboard keys to the UI keyboard for now, but can reduce that to just some UI keys and 1.5 octaves of keys instead if the answer is no.

Reply:
`IPT_KEYBOARD` and `IPT_KEYPAD` inputs are special in a couple of ways:
* They can be enabled or disabled per emulated device in the input settings menus (this is particularly useful when you have a system with multiple keyboards, e.g. a computer with a system keyboard and a serial terminal connected)
* If any device in the system has `IPT_KEYBOARD` inputs, MAME will start with UI controls disabled and allow UI controls to be enabled/disabled by the user.

Apart from that, there isn’t anything magical about them.  They can be assigned like any other digital inputs.

Reply:
All right, I've tried one more layout that provides most panel switches and one set of keyboard keys. Description in comment in top of file. PTAL